src/components/Footer.tsx

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import classNames from 'classnames';
+import { useEffect, useState } from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  setVisibleTodos: (arg: Todo[]) => void;
+  visibleTodos: Todo[];
+};
+
+export const Footer = ({ todos, setVisibleTodos, visibleTodos }: Props) => {
+  const [filterType, setFilterType] = useState('all');
+
+  useEffect(() => {
+    switch (filterType) {
+      case 'all':
+        setVisibleTodos(todos);
+        break;
+
+      case 'completed':
+        setVisibleTodos(todos.filter((todo) => todo.completed));
+        break;
+
+      case 'active':
+        setVisibleTodos(todos.filter((todo) => !todo.completed));
+        break;
```

Comment:
Create an enum for types. It is bad to use string values, as a mistake in one letter can break the whole code.